kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentFactoryTest.java

```diff
@@ -122,6 +127,38 @@ public void getSegmentSourcesMap() {
         commonEvaluateMap(retrieved, segment);
     }
 
+    @Test
+    public void getSegmentSourcesMapCompiled() {
+        final Segment segment = MINING_MODEL.getSegmentation().getSegments().get(0);
+        final List<KiePMMLModel> nestedModels = new ArrayList<>();
+        final String modelName = segment.getModel().getModelName();
+        final String sanitizedPackageName = getSanitizedPackageName(PACKAGE_NAME + "."
+                                                                      + segment.getId() + "."
+                                                                      + modelName);
+        final String sanitizedClassName = getSanitizedClassName(modelName);
+        final String expectedGeneratedClass = String.format(PACKAGE_CLASS_TEMPLATE, sanitizedPackageName, sanitizedClassName);
+        final HasKnowledgeBuilderMock hasKnowledgeBuilderMock = new HasKnowledgeBuilderMock(KNOWLEDGE_BUILDER);
+        try {
+            hasKnowledgeBuilderMock.getClassLoader().loadClass(expectedGeneratedClass);
+            fail("Expecting class not found: " + expectedGeneratedClass);
+        } catch (Exception e) {
+            assertTrue(e instanceof ClassNotFoundException);
+        }
+        final Map<String, String> retrieved = KiePMMLSegmentFactory.getSegmentSourcesMapCompiled(PACKAGE_NAME,
+                                                                                                 DATA_DICTIONARY,
+                                                                                                 TRANSFORMATION_DICTIONARY,
+                                                                                                 segment,
+                                                                                                 hasKnowledgeBuilderMock,
+                                                                                                 nestedModels);
+        commonEvaluateNestedModels(nestedModels);
+        commonEvaluateMap(retrieved, segment);
+        try {
+            hasKnowledgeBuilderMock.getClassLoader().loadClass(expectedGeneratedClass);
+        } catch (Exception e) {
+            fail("Expecting class to be loaded, but got: " + e.getClass().getName() + " -> " + e.getMessage());
```

Comment:
I think it would be better to wrap the exception into RuntimeException and rethrow. Or it is possible to add it to method definition. My main concern is to see more details of underlying exception like stack trace.

Reply:
@jiripetrlik 
done